ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/utils/Keys.kt

```diff
@@ -0,0 +1,37 @@
+/**
+ * Contains functions to determine the type of key during encoding and writing.
+ */
+
+package com.akuleshov7.ktoml.utils
+
+/**
+ * Returns true if the specified key is a bare key, a key containing only
+ * alphanumeric, underscore, and hyphen characters.
+ *
+ * @return `true` if this key is a bare key.
+ */
+internal fun String.isBareKey() =
+        all {
+            it in 'A'..'Z' ||
+                    it in 'a'..'z' ||
+                    it in '0'..'9' ||
+                    it in "-_"
```

Comment:
May be better to make a set of symbols instead of several OR (||) checks:
`it in (('A'..'Z') + ('a'..'z') + ('0'..'9') + "-_")`

Reply:
Good point. Could even add a toString at the end and make it a static property